pandas/tools/tile.py

```diff
@@ -249,16 +259,24 @@ def _format_levels(bins, prec, right=True,
             if a != b and fa == fb:
                 raise ValueError('precision too low')
 
-            formatted = '(%s, %s]' % (fa, fb)
+            if time_data:
```

Comment:
look at ``pandas/core/algorithms.py/factorize`` for how we handle the differing dtypes. we want to replicate that type of logic (what you are doing is fine, but the code will be more smooth).